xtractmime/_utils.py

```diff
@@ -0,0 +1,269 @@
+from struct import unpack
+from typing import Tuple, Union
+
+from xtractmime import _is_match_mime_pattern
+from xtractmime._patterns import (
+    ARCHIVE_PATTERNS,
+    AUDIO_VIDEO_PATTERNS,
+    FONT_PATTERNS,
+    IMAGE_PATTERNS,
+)
+
+SAMPLE_RATES = (44100, 48000, 32000)
+MP25_RATES = (
+    0,
+    8000,
+    16000,
+    24000,
+    32000,
+    40000,
+    48000,
+    56000,
+    64000,
+    80000,
+    96000,
+    112000,
+    128000,
+    144000,
+    160000,
+)
+MP3_RATES = (
+    0,
+    32000,
+    40000,
+    48000,
+    56000,
+    64000,
+    80000,
+    96000,
+    112000,
+    128000,
+    160000,
+    192000,
+    224000,
+    256000,
+    320000,
+)
+
+
+def is_mp4_signature(input_bytes: bytes) -> bool:
+    input_size = len(input_bytes)
+    if input_size < 12:
+        return False
+
+    box_size = unpack(">I", input_bytes[0:4])[0]
+
+    if input_size < box_size or box_size % 4:
+        return False
+
+    if input_bytes[4:8] != b"ftyp":
+        return False
+
+    if input_bytes[8:11] == b"mp4":
+        return True
+
+    bytes_read = 16
+    while bytes_read < box_size:
+        if input_bytes[bytes_read : bytes_read + 3] == b"mp4":
+            return True
+        bytes_read += 4
+
+    return False
+
+
+def parse_vint_number_size(input_bytes: memoryview) -> int:
+    """Returns an integer value by which the index in the current input bytes of a
```

Comment:
(Nitpick alert) two suggestions about this and the other docstrings in the file, from [PEP 257 -- Docstring Conventions](https://www.python.org/dev/peps/pep-0257/):
* "The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ..."."
* "Unless the entire docstring fits on a line, place the closing quotes on a line by themselves"